internal/gatewayapi/filters.go

```diff
@@ -836,27 +835,13 @@ func (t *Translator) processRequestMirrorFilter(
 
 	mirrorEndpoints, _ := t.processDestEndpoints(mirrorBackendRef, filterContext.ParentRef, filterContext.Route, resources)
 
-	// Only add missing mirror destinations
-	for _, mirrorEp := range mirrorEndpoints {
-		var found bool
-		if filterContext.Mirror != nil {
-			for _, mirror := range filterContext.Mirror.Endpoints {
-				if mirror != nil {
-					if mirror.Host == mirrorEp.Host && mirror.Port == mirrorEp.Port {
-						found = true
-					}
-				}
-			}
-		}
-		if !found {
-			if filterContext.Mirror == nil {
-				filterContext.Mirror = &ir.RouteDestination{
-					Name: fmt.Sprintf("%s-mirror", irRouteDestinationName(filterContext.Route, filterContext.RuleIdx)),
-				}
-			}
-			filterContext.Mirror.Endpoints = append(filterContext.Mirror.Endpoints, mirrorEp)
-		}
+	newMirror := &ir.RouteDestination{
+		Name: fmt.Sprintf("%s-mirror-%d", irRouteDestinationName(filterContext.Route, filterContext.RuleIdx), filterIdx),
 	}
+	filterContext.Mirror = append(filterContext.Mirror, newMirror)
+	// Get the index of the last mirror added
```

Comment:
why do we need L842-L844, can't we just assign `Endpoints` on L839 when creating `RouteDestination` ?